locator/token_metadata.cc

```diff
@@ -904,11 +913,15 @@ future<> token_metadata_impl::update_pending_ranges(
 }
 
 size_t token_metadata_impl::count_normal_token_owners() const {
+    return _nr_normal_token_owners;
+}
+
+void token_metadata_impl::update_normal_token_owners() {
     std::set<inet_address> eps;
```

Comment:
As I mentioned in https://github.com/scylladb/scylla/issues/11128, using a `std::set` here is wasteful since insert to it is done in O(log(n)) to maintain ordering (using a balanced tree).  Since we're interested just in the size of the unique set, `std::unordered_set` would excel in this job since insertion to it is typically O(1) using a hash table.

This could be done within the scope of this PR and is easy and safe to backport.

Reply:
OK. Done.